File: treebolicFilesServices/src/main/java/org/treebolic/files/service/TreebolicFilesMessengerService.java
/*
 * Copyright (c) Treebolic 2019. Bernard Bou <1313ou@gmail.com>
 */

package org.treebolic.files.service;

import org.treebolic.services.TreebolicMessengerService;

import androidx.annotation.NonNull;

/**
 * Treebolic Files messenger bound service
 *
 * @author Bernard Bou
 */
public class TreebolicFilesMessengerService extends TreebolicMessengerService
{
	/**
	 * Constructor
	 */
	public TreebolicFilesMessengerService()
	{
		super();
	}

	@Override
	public void onCreate()
	{
		super.onCreate();
		this.factory = new ModelFactory(this);
	}

	@SuppressWarnings("SameReturnValue")
	@NonNull
	@Override
	public String getUrlScheme()
	{
		return "directory:";
	}
}

File: treebolicFilesServices/src/main/java/org/treebolic/files/service/TreebolicFilesMessengerService.kt
/*
 * Copyright (c) 2023. Bernard Bou
 */
package org.treebolic.files.service

import org.treebolic.services.TreebolicMessengerService

/**
 * Treebolic Files messenger bound service
 *
 * @author Bernard Bou
 */
class TreebolicFilesMessengerService : TreebolicMessengerService() {

    override fun onCreate() {
        super.onCreate()
        factory = ModelFactory(this)
    }

    override val urlScheme: String
        get() = "directory:"
}
